src/controllers/ArtsController.js

```diff
@@ -417,6 +419,235 @@ class ArtsController {
       return res.status(response.code).json(response);
     }
   }
+
+  /**
+   * @static
+   * @desc POST /api/v1/arts/:artId/like
+   * @param {object} req
+   * @param {object} res
+   * @memberof ArtsController
+   * @returns {object} successful like
+   */
+  static async likeArticle(req, res) {
+    try {
+      const { artId } = req.params;
+      const { id } = req.verifyUser;
+
+      const like = await Like.findOrCreate({
+        where: {
+          artId,
+          userId: id
+        },
+      })
+        .spread((user, created) => {
+          user.get({ plain: true });
+          return created;
+        });
+
+      if (!like) {
+        const response = new Response(
+          'Bad Request',
+          400,
+          'You have already liked this article',
+        );
+        return res.status(response.code).json(response);
+      }
+
+      LikeUnlike.like(artId);
+
+      const checkIfDisliked = await Like.findOne({
```

Comment:
What's the difference between this query and the other one? At what point will this code run?

Reply:
There are two functionalities implemented; `Like` and `Dislike`. Apparently, a user cannot dislike what he has already liked, neither can he like what he already disliked. So, in this case, I'm just checking if the user's details exist with specified art id on the Dislike model before completing the function.

At the point the user's details alongside the art id can be found in the Dislike model, I delete that record and complete the like for the user.